core/src/main/java/com/sequenceiq/cloudbreak/converter/ClusterTemplateStringToClouderaApiClusterTemplateConverter.java

```diff
@@ -0,0 +1,139 @@
+package com.sequenceiq.cloudbreak.converter;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplate;
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.cloudera.api.swagger.model.ApiClusterTemplateHostInfo;
+import com.cloudera.api.swagger.model.ApiClusterTemplateHostTemplate;
+import com.cloudera.api.swagger.model.ApiClusterTemplateInstantiator;
+import com.cloudera.api.swagger.model.ApiClusterTemplateRoleConfigGroup;
+import com.cloudera.api.swagger.model.ApiClusterTemplateService;
+import com.cloudera.api.swagger.model.ApiClusterTemplateVariable;
+import com.cloudera.api.swagger.model.ApiProductVersion;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import com.sequenceiq.cloudbreak.service.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.util.JsonUtil;
+
+@Component
+public class ClusterTemplateStringToClouderaApiClusterTemplateConverter extends AbstractConversionServiceAwareConverter<String, ApiClusterTemplate> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterTemplateStringToClouderaApiClusterTemplateConverter.class);
+
+    @Override
+    public ApiClusterTemplate convert(String source) {
+        try {
+            JsonNode clusterTemplateJson = JsonUtil.readTree(source);
+            ApiClusterTemplate apiClusterTemplate = new ApiClusterTemplate();
```

Comment:
Would it be possible to read it directly to ApiClusterTemplate?

Reply:
Probably yes, will be addressed in next PRs

Reply:
why not now, this feels a bit ugly

Reply:
okay, i'll look into it